### PR TITLE
Update docs for `guard`

### PR DESCRIPTION
**Pull request by JamieWohletz**
This PR expands the docs for `guard` a little bit, adding `import`s and fixing the `factors` function in the example provided.

Lemme know if you'd like me to squash my commits!

### CODE REVIEW

File: src/Control/MonadZero.purs [hunk 1]
```diff
@@ -39,15 +39,22 @@ instance monadZeroArray :: MonadZero Array
...
 -- | factors n = do
--- |   a <- 1 .. n
--- |   b <- 1 .. a
+-- |   a <- 1..n
+-- |   b <- a..n
 -- |   guard $ a * b == n
--- |   pure a
+-- |   pure [a, b]
 -- | ```
+-- | In this example, a step of the array comprehension will fail with
```
Comment (paf31) [inline src/Control/MonadZero.purs]: I wouldn't mention `empty` here, since readers of the documentation won't necessarily be reading the source code.

Comment (JamieWohletz) [inline src/Control/MonadZero.purs]: got it, will update tonight, thanks!

**Comment by JamieWohletz** [comment]
@paf31 do these changes work?

**Comment by paf31** [comment]
Yes, looks great, thanks!

**Pull request merged**